io/net/net_data.py

```diff
@@ -88,30 +88,38 @@ def testbigping(self):
         msg = "ip addr show  | grep %s | grep -oE '[^ ]+$'" % self.peer
         cmd = "ssh %s %s" % (self.peer, msg)
         errors = []
-        self.peer_interface = process.system_output(cmd, shell=True).strip()
+        peer_interface = process.system_output(cmd, shell=True).strip()
         mtuval = process.system_output("ip link show %s" % self.interface,
                                        shell=True).split()[4]
         for mtu in self.mtu_list:
-            self.log.info("trying with mtu %s" % (mtu))
+            flag = 0
+            self.log.info("trying with mtu %s", mtu)
             # ping the peer machine with different maximum transfers unit sizes
             # and finally set maximum transfer unit size to 1500 Bytes
             msg = "ssh %s \"ip link set %s mtu %s\"" % (self.peer,
-                                                        self.peer_interface,
+                                                        peer_interface,
                                                         mtu)
-            process.system(msg, shell=True)
+            if process.system(msg, shell=True, ignore_status=True):
+                self.log.debug("setting mtu value %s in peer failed", mtu)
+                flag = 1
```

Comment:
I always find this `if process.system()` confusing and the reason is because it returns the exit code and not True/False. Testing the exit code with an `if` gives the impression of wrong logics because `bool(0)` is `False` and `bool(1)` is `True`. It's not your problem, it's just how it works. But to avoid such weirdness, I'd use something like:
```python
try:
    process.system(msg, shell=True)
except process.CmdError:
    self.log.debug("setting mtu value %s in peer failed", mtu)
else:
    mtu_set = True
```
Anyway, that's just me thinking out loud. The most important part here is to control the `mtu_set` with `True` or `False`.